docs/proposals/13-automated-seed-management.md

```diff
@@ -263,18 +283,66 @@ ManagedSeedSets are reconciled by a new *managed seed set controller* in `garden
+
+Note that unlike a StatefulSet, an `OnDelete` update strategy is not supported.
+
+**Partitions**
+
+The `RollingUpdate` update strategy can be partitioned, by specifying a `.spec.updateStrategy.rollingUpdate.partition`. If a partition is specified, only ManagedSeeds and Shoots with ordinals greater than or equal to the partition will be updated when any of the ManagedSeedSet's templates is updated. All remaining ManagedSeeds and Shoots will not be updated. If a ManagedSeedSet's `.spec.updateStrategy.rollingUpdate.partition` is greater than the largest ordinal number in use by a replica, updates to its templates will not be propagated to its replicas (but newly added replicas may still use the updated templates depending on the partition value).
+
+#### Keeping Track of Revision History and Performing Rollbacks
 
-A ManagedSeedSet, similarly to a `ReplicaSet`, does not manage updates to its replicas in any way. In the future, we might introduce ManagedSeedDeployments, a higher-level concept that manages ManagedSeedSets and provides declarative updates to ManagedSeeds along with other useful features, similarly to a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), or perhaps a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/). An implementation that is more like a StatefulSet rather than a Deployment would be rolling seed by seed, stopping on rollout-failures (see [Issue #87](https://github.com/gardener/gardener/issues/87)), etc.
+Similarly to a StatefulSet, the ManagedSedSet controller uses [ControllerRevisions](https://pkg.go.dev/k8s.io/api/apps/v1#ControllerRevision) to keep track of the revision history, and `controller-revision-hash` labels to maintain an association between a ManagedSeed or a Shoot and the concrete template revisions based on which they were created or last updated. These are used for the following purposes:
 
-There is an important difference between seeds and pods or nodes in that seeds are more "heavyweight" and therefore updating a set of seeds by introducing new seeds and moving shoots to them tends to be much more complex, time-consuming, and prone to failures compared to updating the seeds "in place". Furthermore, updating seeds in this way depends on a mature implementation of [GEP-7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md), which is not available right now. Therefore, ManagedSeedDeployments are not described in detail here, as properly dealing with the challenges mentioned requires a separate GEP.
+* During an update, determine which replicas are still not on the latest revision and therefore should be updated.
+* Display the revision history of a ManagedSedSet via `kubectl rollout history`.
+* Roll back all ManagedSedSet replicas to a specific revision via `kubectl rollout undo`
```

Comment:
No, I don't think it will work directly. [Documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#rollout) says explicitly that valid resource types are deployments, daemonsets, and statefulsets, and I don't see how they could implement it to support custom types out of the box. But I think it should be possible to support rollout commands for our types via a [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/). I will update the document to properly mention this.